stdlib/asm/crypto/stark/public_inputs.masm

```diff
@@ -81,13 +81,153 @@ export.load
         end
 
         # Absorb the last digest and pad with zeros
-        adv_loadw
-        swapw
-        exec.constants::zeroize_stack_word
+        exec.load_base_store_extension_word
         hperm
     end
 
     exec.rpo::squeeze_digest
     movup.4 drop
     movup.4 drop
 end
+
+#! Computes the address where the public inputs are to be stored.
+#!
+#! In order to be able to call `arithmetic_circuit_eval`, we need to layout the inputs to
+#! the constraint evaluation circuit in a contiguous region of memory (called `READ` section
+#! in the ACE chiplet documentation) right before the region of memory storing the circuit
+#! description (called `EVAL` section in the ACE chiplet documentation).
+#! As the number of public inputs is a per-instance parameter, while the sizes of the OOD
+#! evaluation frames and the number of auxiliary random values are fixed, we can lay out
+#! the public inputs right before the auxiliary random values and OOD evaluations.
+#! Hence the address where public inputs are stored is computed using a negative offset
+#! from the address where the OOD are stored.
+#!
+#! Input: [...]
+#! Output: [...]
+proc.compute_public_inputs_address
```

Comment:
```suggestion
proc.compute_and_store_public_inputs_address
```
since IMO `compute_*` suggests that the public inputs address is returned

Reply:
Agreed and changed